GlimpseImputationPipeline/Glimpse2Imputation.wdl

```diff
@@ -429,3 +430,46 @@ task SelectResourceParameters {
         Int request_n_cpus = read_int("n_cpus_request.txt")
     }
 }
+
+task CombineCoverageMetrics
+{
+    input {
+        Array[File] cov_metrics
+    }
+
+    command <<<
+        set -euo pipefail
+
+        cov_files=( ~{sep=" " cov_metrics} )
+
+        for i in "${!cov_files[@]}"; do
+            if [ $i -eq 0 ]; then
+                n_skip=1
+                echo 'Chunk' > chunk_col.txt
+            else
+                n_skip=2
+            fi
+            # glimpse coverage metrics are formatted to be human readable in a command line, not machine readable or consistent.  ie, number of tabs 
+            # are variable between columns depending on length of sample names, odd things like that.  We want these to be machine readable tables, 
+            # so need to fix this.
+            zcat ${cov_files[$i]} | tail -n +$((n_skip + 1)) | sed s/%//g | sed s/"No data"/"No data pct"/g | sed s/\\t\\t/\\t/g >> cov_file.txt
+            n_lines_cov=$(< cov_file.txt wc -l)
+            n_lines_chunk=$(< chunk_col.txt wc -l)
+            n_lines_out=$((n_lines_cov-n_lines_chunk))
+            echo 'n_lines_out=' ${n_lines_out}
+            echo ${cov_files[$i]}
+            { yes ${i} || :; } | head -n ${n_lines_out} >> chunk_col.txt
+        done
+
+        paste chunk_col.txt cov_file.txt > coverage_metrics.txt
+
+    >>>
+
+    runtime {
+        docker: "ubuntu:24.04"
+    }
+
+    output {
+        File coverage_metrics="coverage_metrics.txt"
```

Comment:
I tried to name all output files with the sample name (or `basename`) in the filename, in order to make metrics storage and debugging slightly easier. If you want you can implement that here by passing `basename` to this task and changing the output file name to `~{basename}.coverage_metrics.txt` (also in line 464 of course)

Reply:
good call, will add basename